tests/test_highlighting.py

```diff
@@ -34,13 +39,16 @@ class MyLexer(RegexLexer):
 
 
 class MyFormatter(HtmlFormatter[str]):
-    def format(self, tokensource, outfile):
-        for tok in tokensource:
-            outfile.write(tok[1])
+    def format(
+        self,
+        tokensource: Iterable[tuple[_TokenType, str]],
+        outfile: TextIO,
+    ) -> None:
+        outfile.writelines(tok[1] for tok in tokensource)
 
 
 class ComplainOnUnhighlighted(PygmentsBridge):
-    def unhighlighted(self, source):
+    def unhighlighted(self, source: str) -> Never:
```

Comment:
I don't actually know what type `source` is

Reply:
Good catch, it seems uses of `unhighlighted(` were removed some time ago:

* b29226a0609d5803afd51c01f6d522bd30592dbf ("Drop features and APIs deprecated in 1.8")
* db1cf80a69778d882f5c39b8966dc9fd5af256ac ("Make pygments unconditional, it is required by setup.py anyway.")
* 61098a0ae2e696a804459d36bd74ca57db76eda5 ("In code blocks, when the selected lexer fails, display line numbers nevertheless if configured.")